client/forest-manager/src/prove.rs

```diff
@@ -0,0 +1,253 @@
+use storage_hub_infra::types::{Leaf, Proven};
```

Comment:
Since this isn't in the context of a runtime, `utils.rs` is actualy just utility functions, not business logic.

The `prove.rs` file is a big file that will potentially contain more unit tests, evolve over time and also emphasizes the core function that drives our proving scheme which is used across all `ForestStorage` implementations (soon we will have rocksdb as well which will use this function).